classes/form/guestlink_access_form.php

```diff
@@ -0,0 +1,97 @@
+
+require_once($CFG->dirroot . '/mod/bigbluebuttonbn/locallib.php');
+require_once($CFG->libdir . '/formslib.php');
+require_once($CFG->dirroot . '/mod/bigbluebuttonbn/mod_form.php');
+
+/**
+ * Moodle class for mod_form.
+ *
+ * @copyright Kevin Pham <kevinpham@catalyst-au.net>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class guestlink_access_form extends \moodleform {
+
+    /**
+     * Define (add) particular settings this activity can have.
+     *
+     * @return void
+     */
+    public function definition() {
+        global $CFG, $DB, $OUTPUT, $PAGE;
+        $mform = &$this->_form;
+
+        // Guest Access URL.
+        $inviteparticipantselement = $mform->createElement('text', 'guestlinkurl', get_string('view_guestlink_label', 'bigbluebuttonbn'), ['readonly' => true, 'size' => 60]);
+        $mform->setType('guestlinkurl', PARAM_TEXT);
+        $mform->setDefault('guestlinkurl', $this->_customdata['guestlinkurl']);
+
+        $inviteparticipantsgroup = [];
+        $inviteparticipantsgroup[] =& $inviteparticipantselement;
+        $inviteparticipantsgroup[] =& $mform->createElement('button', 'guestlinkurl_copy', get_string('mod_guestlink_access_form_copy', 'bigbluebuttonbn'));
+        $mform->addGroup($inviteparticipantsgroup, 'inviteparticipantsgroup', get_string('view_guestlink_label', 'bigbluebuttonbn'), array(' '), false);
+
+        // Access Code.
+        $accesscodeelement = $mform->createElement('text', 'password', '', [
+            'readonly' => true,
+            'size' => 29,
+            'placeholder' => get_string('view_guestlink_password_no_password_set', 'bigbluebuttonbn')
+        ]);
+        $mform->setType('password', PARAM_TEXT);
+        $mform->setDefault('password', $this->_customdata['guestlinkpassword'] ?? '');
+
+        $accesscodegroup = [];
+        $accesscodegroup[] =& $accesscodeelement;
+        if (!empty($this->_customdata['guestlinkchangepassenabled'])) {
+            $accesscodegroup[] =& $mform->createElement('button', 'password_clear', get_string('mod_guestlink_access_form_clear', 'bigbluebuttonbn'));
+        }
+        if (!empty($this->_customdata['guestlinkchangepassenabled'])) {
+            $accesscodegroup[] =& $mform->createElement('button', 'password_change', get_string('mod_guestlink_access_form_generate', 'bigbluebuttonbn'));
+        }
+        $accesscodegroup[] =& $mform->createElement('button', 'password_copy', get_string('mod_guestlink_access_form_copy', 'bigbluebuttonbn'));
+        $mform->addGroup($accesscodegroup, 'accesscodegroup', get_string('view_guestlink_password_label', 'bigbluebuttonbn'), array(' '), false);
+
+        // Expires At.
+        $defaultdateoptions = [
+           'optional' => true
+        ];
+        $expiresatelement = $mform->createElement('date_time_selector', 'guestlinkexpiresat', '', $defaultdateoptions);
+        $mform->setType('guestlinkexpiresat', PARAM_RAW);
+
+        $accesscodegroup = [];
```

Comment:
Funny thing, adding the linebreak helped expose this. Its tiny, but the varname is re-used here, better to make it correct to the new group.

Reply:
Good catch